src/index.tsx

```diff
@@ -7,6 +7,49 @@ import { StrictMode } from "react";
 import { createRoot } from "react-dom/client";
 import App from "./App";
 import reportWebVitals from "./reportWebVitals";
+import { registerSW } from "virtual:pwa-register";
+import { flags } from "./flags";
+import { baseUrl } from "./base";
+
+if (flags.pwa) {
+  registerSW({
+    immediate: true,
+    // Cache runtime resources on first load.
+    // See https://github.com/GoogleChromeLabs/pwa-wp/issues/180.
+    onRegisteredSW(_, registration) {
+      // Inject webmanifest.
+      const link = document.createElement("link");
+      link.rel = "manifest";
+      link.href = `${baseUrl}manifest.webmanifest`;
+      document.head.appendChild(link);
+
+      if (registration) {
+        registration.onupdatefound = function () {
+          const installingWorker = registration?.installing;
+          if (installingWorker) {
+            installingWorker.onstatechange = function () {
+              if (
+                installingWorker.state === "activated" &&
+                navigator.serviceWorker.controller
+              ) {
+                const urlsToCache = [
+                  location.href,
+                  ...performance
+                    .getEntriesByType("resource")
+                    .map((r) => r.name),
+                ];
+                installingWorker.postMessage({
+                  type: "CACHE_URLS",
+                  payload: { urlsToCache },
+                });
+              }
+            };
+          }
+        };
+      }
+    },
+  });
+}
```

Comment:
Do we need an `else` here similar to the sim?
Scenario to test: use flag, update app, then don't use flag and should see latest

Reply:
61213d240c8931b5636bbfb739fcd7f3fa74cda8